tensorboard/webapp/widgets/linked_time_fob/linked_time_fob_controller_component.ts

```diff
@@ -194,4 +194,31 @@ export class LinkedTimeFobControllerComponent {
     // No fobs can pass the lowest step in this graph.
     return 0;
   }
+
+  stepTyped(fob: Fob, step: number) {
+    let newLinkedTime = this.linkedTime;
+    if (!newLinkedTime.end) {
```

Comment:
It seems like this entire branch of logic (the if block, the extra emit() call, the early return) only exists to narrow the `end` field to be non-null for the rest of the function. Its sort of a compiler workaround. That's unfortunate.

Alternatively:
Modify line 208 to check that newLinkedTime.end !== null (or !newLinkedTime.end, depending if you are allowing 0, too).
And modify line 215 so that newLinkedTime.end = {step};

Then you don't need this if block at all.

Reply:
handled in summary comment.